SLE-851: Handle simple and complex timeout configuration

thahnen opened the pull request:
With that SonarLint can now handle simple (like "3" for "3 minutes") and complex (like "PT4H" for "4 hours") timeout configurations again.

nquinquenel commented on org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/utils/DurationUtils.java:
```diff
@@ -0,0 +1,47 @@
+    try {
+      return Duration.ofMinutes(Integer.parseInt(property));
+    } catch (NumberFormatException ignored) {
+      return Duration.parse(property);
```
There is a `DateTimeParseException` thrown in case it's wrong, shouldn't it be safer to catch it and return `null`?

thahnen replied:
I re-used the old logic that got lost (see [ticket comment](https://sonarsource.atlassian.net/browse/SLE-851?focusedCommentId=604251)) from SLCORE.

I think you're right. I will change it that it is logged and then returns `null`. I wouldn't fail hard in this case but for any means necessary the user should be informed the input was garbage :smile:

nquinquenel reviewed (approved):
LGTM

thahnen committed:
SLE-851: Handle simple and complex timeout configuration

With that SonarLint can now handle simple (like "3" for "3 minutes") and complex (like "PT4H" for "4 hours") timeout configurations again.
